crytic_compile/platform/hardhat.py

```diff
@@ -226,6 +210,9 @@ def is_supported(target: str, **kwargs: str) -> bool:
 
         # If there is both foundry and hardhat, foundry takes priority
         if os.path.isfile(os.path.join(target, "foundry.toml")):
+            LOGGER.info(
+                "foundry.toml found, ignoring hardhat. To force hardhat, use `--compile-force-framework hardhat`"
+            )
```

Comment:
I'm thinking that this may print in unexpected places (e.g. foundry-only projects, or slither-doctor), have you verified if that's the case?.

I think it'd be better to move this "override" logic outside of each platform and into the code that evaluates supported platforms. Maybe we can have a "priority" of them and pick the best one out of all the supported ones, or evaluate one by one in the preferred order and stop when a supported platform is found?

Reply:
You're right. `is_supported` is called a number of places so this won't work as it stands.

Reply:
i removed the log